_posts/2022-04-13-webhttp.markdown

```diff
@@ -0,0 +1,148 @@
+
+First we'll want to enable help for our contract. Change the AddServiceWebEndpoint call to be:
+
+```csharp
+builder.AddServiceWebEndpoint<WebApi, IWebApi>("api", behavior =>
+{
+    behavior.HelpEnabled = true;
+});
+```
+
+Note that this is also how you would configure the WebHttpBehavior for any other reason as well. We'll also need another Nuget package. Add it by running: `dotnet add package Swashbuckle.AspNetCore.SwaggerUI`. Back in the startup we'll need to add something else to services:
+
+```csharp
+builder.Services.AddSingleton(new SwaggerOptions());
+```
+
+And some middleware:
+
+```csharp
+app.UseMiddleware<SwaggerMiddleware>();
+app.UseSwaggerUI();
+```
+
+Now CoreWCF will try and intuit as much information as it can about your API for documentation, but sometimes you will have to feed it information directly. This will be done with attributes. Let's add the our first attribute to the service contract:
+
+```csharp
+[OpenApiBasePath("/api")]
+```
+
+This is the same address that the endpoint was registered under in the startup. With all of that done go ahead and run the app, and then navigate to <http://localhost:8080/swagger/> in a browser. There you will see Swagger UI documenting the API, and you can even exercise the API from that page. Let's cover some other documentation related attributes.
+
+If you want to hide something in the documentation you can do so like:
+
+```csharp
+[OpenApiHidden]
+```
+
+If you want something to be under a particular tag you can do so like:
+
+```csharp
+[OpenApiTag("")]
+```
+
+If you want to add a description and/or summary to an operation you can do so like:
+
+```csharp
+[OpenApiOperation(Description = "", Summary = "")]
+```
+
+If you want to add a description and/or supported content type to a parameter you can do so like:
+
+```csharp
+[OpenApiParameter(ContentTypes = new[] { ""}, Description = "")]
+```
+
+Note that all parameters are considered required unless a default value is provided. If you want to add a description to a data contracts property and/or mark it as required you can do so like:
+
+```csharp
+[OpenApiProperty(Description = "", Required = true)]
+```
+
+If you want to add a possible response to an operation you can do so like:
+
+```csharp
+[OpenApiResponse(ContentTypes = new[] { "" }, Description = "", StatusCode = HttpStatusCode.OK, Type = typeof(bla))]
+```
+
+Note that more than one OpenApiResponse attribute can be added to an operation. Additionally a default response will be added for you if you don't specify one.
+
```

Comment:
It would be nice to see the attributes used in context. I think doing it in the blog post itself will end up looking messy and like a wall of code. Maybe add a sentence saying you can see all of these attributes in use in the sample and provide a link to the sample now it's committed to the main branch. Use a commit specific url though and not one pointing to the head of main as at some point the sample will move to their own repo and a "main" base url will break in the future, but a commit hash based url will be forever stable.